ucf_exchange_client/app.py

```diff
@@ -14,23 +15,23 @@ def __init__(self, name, default_handlers=True):
 
         self.orderid = 0
         self.orders = dict()
-        
+
         # TODO add a clock here, for scheduled functions as well as if strategies need
         # access to a clock.
         self.clock = None
 
     def run(self, host, port):
         """Run the strategy against a connected exchange."""
-        sock = get_conn(host, port)
-        while not sock.closed:
-            # TODO: figure out how to run scheduled functions within this event loop,
-            # since get_msg is probably blocking. (We can make it non-blocking to make
-            # stuff easier).
-            msg = msg_from_json(get_msg(sock))
-            for outmsg in self._handle(msg):
-                # TODO if we're making orders then we should
-                # handle the orderid stuff automatically.
-                sock.write(msg_to_json(msg._asdict()))
+        asyncio.get_event_loop().run_until_complete(self._run(host, port))
+
+    async def _run(self, host, port):
+        """Sets up the connection and handles incoming messages."""
+        self.conn = await connection.create(host, port)
+        while True:
+            msg = await self.conn.read()
+            resp = await self._handle(msg)
```

Comment:
I don't think you need to await this? It's not an async operation, and `_handle` isn't async-def'd anyway.